src/SourceBuild/tarball/content/prep.sh

```diff
@@ -75,20 +71,35 @@ if [ -d $SCRIPT_ROOT/.dotnet ]; then
 fi
 
 # Read the archive text file to get the archives to download and download them
-while read -r line; do
-    if [[ $line == *"Private.SourceBuilt.Artifacts"* ]]; then
-        if [ "$downloadArtifacts" == "true" ]; then
-            echo "  Downloading source-built artifacts from $line..."
-            (cd $SCRIPT_ROOT/packages/archive/ && curl --retry 5 -O $line)
-        fi
+sourceBuiltArtifactsTarballUrl="https://dotnetcli.azureedge.net/source-built-artifacts/assets/"
+packageVersionsPath="$SCRIPT_ROOT/eng/Versions.props"
+
+if [ "$downloadArtifacts" == "true" ]; then
+    echo "  Looking for source-built artifacts to download..."
+    artifactsVersionLine=`grep -m 1 '<PrivateSourceBuiltArtifactsPackageVersion>' "$packageVersionsPath" || :`
+    versionPattern="<PrivateSourceBuiltArtifactsPackageVersion>(.*)</PrivateSourceBuiltArtifactsPackageVersion>"
+    if [[ $artifactsVersionLine =~ $versionPattern ]]; then
+        artifactsUrl="${sourceBuiltArtifactsTarballUrl}${artifactsBaseFileName}.${BASH_REMATCH[1]}.tar.gz"
+        echo "  Downloading source-built artifacts from $artifactsUrl..."
+        (cd $SCRIPT_ROOT/packages/archive/ && curl --retry 5 -O $artifactsUrl)
+    else
```

Comment:
Does it make sense to extract a function for grepping + pulling the file based on a props key?

Reply:
I waffled on this.  I encapsulated it within a function.  It gets a little more complex because the two invocations were slightly different.